### ISSUE TITLE
Webpack: Note about restarting webpack-dev-server on config change

### ISSUE THREAD

**Issue by MaoShizhong** [TheOdinProject/curriculum#28766]
### Checks

- [X] This is not a duplicate of an existing issue (please have a look through our [open issues list](https://github.com/TheOdinProject/curriculum/issues) to make sure)
- [X] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
- [ ] Would you like to work on this issue?

### Describe your suggestion

A somewhat common issue learners run into is not realising that the webpack config only applies to webpack-dev-server when you start it. If you change the webpack config file, the dev server has already been set up with the old config so you need to kill it (`ctrl + c`) and restart it to apply the new config.

## Acceptance criteria

- [ ] The end of the [webpack-dev-server part of the Webpack lesson](https://www.theodinproject.com/lessons/javascript-webpack#webpack-dev-server) mentions that if you make any changes to `webpack.config.js`, you need to restart the dev server to apply the new config.
  - [ ] Could just be a sentence or two. Perhaps could be a [note box](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md#note-boxes) if you fancy.
  - [ ] Use [`<kbd>` tags](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md#keyboard-shortcuts) to indicate that <kbd>Ctrl</kbd> + <kbd>C</kbd> will kill the dev server.

Please comment below if you'd like to be assigned to this issue. Please do not open a PR unless you have been assigned by a maintainer.

### Path

Ruby / Rails, Node / JS

### Lesson Url

https://www.theodinproject.com/lessons/javascript-webpack#webpack-dev-server

**Comment by thereisnodeveloper** [comment]
Hi, I'm available to work on this!

**Comment by MaoShizhong** [comment]
All yours!

**Comment by thereisnodeveloper** [comment]
I have found a simple solution that automatically restarts webpack when config changes. Should I open a new issue or a pull request for this? I do not see any downsides, and have tested the method.

npm install --save-dev nodemon

In package.json, add/change the script for "start":
"start": "nodemon --watch webpack.config.js ./node_modules/.bin/webpack-dev-server"

**Comment by MaoShizhong** [comment]
Pedagogically, I don't think nodemon is the way forward. Npm scripts are intentionally left until after the Restaurant Page project and the root problem isn't the restarting bit itself, it's not realising it's even needed in the first place given that it watches changes to the bundle to re-bundle (which isn't the same as actually restarting the dev server itself). I'd rather not introduce yet another package to install just for a small feature that is just for convenience and not a core part of the webpack process - people get confused enough having to install all the other things and what they're for. 

In this case, only a small mention that if you change the config file, you need to restart the dev server is needed. If people wish to search for further convenience things, they can absolutely research that independently (some things we should leave to people to research for themselves if they wish for it).
